src/redux/books/booksSlice.js

```diff
@@ -0,0 +1,22 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = [];
+
+const bookSlice = createSlice({
+  name: 'book',
+  initialState,
+  reducers: {
+    add: (state, action) => {
+      state.push(action.payload);
+    },
+    remove: (state, action) => {
+      const newstate = state.filter((item) => item.id !== action.payload);
+      const updateIndex = newstate.map((item, index) => ({ ...item, id: index + 1 }));
+      return updateIndex;
```

Comment:
- KInldy, Since the filter method does not mutate the original array, you need to save the filtered array to work with it later. and updating the index is not necessary so remove line `14 and 15` 👍 
```javascript
removeBook: (state, action) => {
      state.books = state.books.filter((book) => book.id !== action.id)
    },
```